src/DefaultFilter.php

```diff
@@ -0,0 +1,201 @@
+    {
+        return __("Default search filter, applied in addition to the user's search criteria.");
+    }
+
+    public static function getTypeName($nb = 0)
+    {
+        return _n('Default filter', 'Default filters', $nb);
+    }
+
+    public static function getIcon()
+    {
+        return "ti ti-filter";
+    }
+    public static function canCreate()
+    {
+        return static::canUpdate();
+    }
+
+    public function rawSearchOptions()
+    {
+        $tab = parent::rawSearchOptions();
+
+        $tab[] = [
+            'id'            => '4',
+            'table'         => self::getTable(),
+            'field'         =>  'comment',
+            'name'          =>  __('Comments'),
+            'datatype'      =>  'text'
+        ];
+
+        $tab[] = [
+            'id'                 => '5',
+            'table'              => $this->getTable(),
+            'field'              => 'itemtype',
+            'name'               => __('Itemtype'),
+            'datatype'           => 'itemtypename',
+            'itemtype_list'      => 'globalsearch_types',
+            'massiveaction'      => false
+        ];
+
+        return $tab;
+    }
+
+    public function getAdditionalFields()
+    {
+        return [
+            [
+                'name'  => 'itemtype',
+                'label' => __('Itemtype'),
+                'type'           => 'itemtypename',
+                'itemtype_list'      => 'globalsearch_types',
+            ]
+        ];
+    }
+
+    public static function getSearchCriteria($itemtype)
+    {
+        global $DB;
+
+        $default_table = self::getTable();
+        $filter_table = \Glpi\Search\CriteriaFilter::getTable();
+
+        $criteria = [
+            'SELECT' => [
+                "$default_table.*",
+                "$filter_table.search_criteria",
+            ],
+            'FROM' => $default_table,
+            'JOIN' => [
+                $filter_table => [
+                    'FKEY'  => [
+                        $default_table => 'id',
+                        $filter_table => 'items_id',
+                    ],
+                    'AND'   => [
+                        "$filter_table.itemtype" => __CLASS__
+                    ]
+                ]
+            ],
+            'WHERE' => [
+                "$default_table.itemtype" => $itemtype,
+                "NOT" => [
+                    "$filter_table.search_criteria" => null
+                ]
+            ]
+        ];
+
+        $iterator = $DB->request($criteria);
+
+        if ($iterator->count() == 1) {
+            $item = $iterator->current();
+
+            return [
+                'id' => $item['id'],
+                'name' => $item['name'],
+                'comment' => $item['comment'],
+                'search_criteria' => [
+                    'link' => 'AND',
+                    'criteria' => json_decode($item['search_criteria'], true),
+                ]
+            ];
+        }
+        return [];
```

Comment:
It should return `null` when there is nothing to return.
```suggestion
        return null;
```